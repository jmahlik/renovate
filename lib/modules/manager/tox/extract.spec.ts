import { Fixtures } from '../../../../test/fixtures';
import { extractPackageFile } from '.';

describe('modules/manager/tox/extract', () => {
  describe('extractPackageFile()', () => {
    it('returns null for empty', () => {
      expect(extractPackageFile('nothing here')).toBeNull();
    });

    it('extracts dependencies', () => {
      const res = extractPackageFile(Fixtures.get('tox-ini-1.txt'));
      expect(res).toMatchSnapshot({
        deps: [
          { depName: 'base-test-env', currentValue: '==1.0.0' },
          { depName: 'basic1', currentValue: '>=3.1.24' },
          { depName: 'basic2', currentValue: '>=21.3' },
          { depName: 'basic3', currentValue: '<=21.3' },
          { depName: 'basic4', currentValue: '<1.0.0' },
          { depName: 'basic5', currentValue: '>1.0.0' },
          { depName: 'basic6', currentValue: '==1.0.0' },
          { depName: 'basic7', currentValue: '~=1.0.0' },
          { depName: 'single-line', currentValue: '>=2.9.3' },
          { depName: 'comment-deps1', currentValue: '>=2.16' },
          { depName: 'equals1', currentValue: '>=1.19.4' },
          { depName: 'dep-combos1', currentValue: undefined },
          { depName: 'dep-combos2', currentValue: '==1.0.0' },
        ],
      });
    });
  });
});
