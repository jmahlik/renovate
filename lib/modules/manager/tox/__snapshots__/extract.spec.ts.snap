// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/manager/tox/extract extractPackageFile() extracts dependencies 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": "==1.0.0",
      "currentVersion": "1.0.0",
      "datasource": "pypi",
      "depName": "base-test-env",
      "depType": "test",
    },
    Object {
      "currentValue": ">=3.1.24",
      "datasource": "pypi",
      "depName": "basic1",
      "depType": "test",
    },
    Object {
      "currentValue": ">=21.3",
      "datasource": "pypi",
      "depName": "basic2",
      "depType": "test",
    },
    Object {
      "currentValue": "<=21.3",
      "datasource": "pypi",
      "depName": "basic3",
      "depType": "test",
    },
    Object {
      "currentValue": "<1.0.0",
      "datasource": "pypi",
      "depName": "basic4",
      "depType": "test",
    },
    Object {
      "currentValue": ">1.0.0",
      "datasource": "pypi",
      "depName": "basic5",
      "depType": "test",
    },
    Object {
      "currentValue": "==1.0.0",
      "currentVersion": "1.0.0",
      "datasource": "pypi",
      "depName": "basic6",
      "depType": "test",
    },
    Object {
      "currentValue": "~=1.0.0",
      "datasource": "pypi",
      "depName": "basic7",
      "depType": "test",
    },
    Object {
      "currentValue": ">=2.9.3",
      "datasource": "pypi",
      "depName": "single-line",
      "depType": "test",
    },
    Object {
      "currentValue": ">=2.16",
      "datasource": "pypi",
      "depName": "comment-deps1",
      "depType": "test",
    },
    Object {
      "currentValue": ">=1.19.4",
      "datasource": "pypi",
      "depName": "equals1",
      "depType": "test",
    },
    Object {
      "currentValue": undefined,
      "datasource": "pypi",
      "depName": "dep-combos1",
      "depType": "test",
    },
    Object {
      "currentValue": "==1.0.0",
      "currentVersion": "1.0.0",
      "datasource": "pypi",
      "depName": "dep-combos2",
      "depType": "test",
    },
  ],
}
`;
